### PR TITLE
Feature - Core Infrastructure

### PR DESCRIPTION
**Pull request by DeVaukz**
- DO NOT MERGE - DO NOT MERGE - DO NOT MERGE *

This is still a WIP.  I'm PR'ing early so everyone can begin complaining about my programming or how terrible of an idea this is in general.  Just kidding, hopefully.

See the ReadMe for details. (I don't intend for the ReadMe to be merged)

### CODE REVIEW

**Comment by adam4813** [comment]
As of right now we will focus on using glfw for our platform independence
layer, so the platform\* files are not needed until such a time as we move
to more platform specific code.

Other than that it seems like a good start and we'll have to discuss it in
the meeting.

**Comment by DeVaukz** [comment]
GLFW abstracts windowing systems and GL context setup.  It does not abstract minutia such as how to access the cache prefetch intrinsic (if one exits) or whether the current platform is (or will be built) in 64-bit mode.  That is what is being handled here.  More will be added as necessary.  For example:
- Getting the location where user data should be saved.
- Handling aligned allocations (OS X's allocator is 16-byte aligned by default, other platforms may require a manual implementation).

We're going to need to dip into platform-dependent code and configuration no matter how many libraries we add.  Might as well keep it clean from the start.

**Comment by adam4813** [comment]
Ah. Thanks for the expansion and carry on.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,74 @@
...
+
+See Engine/Core/CMakeLists.txt for a template.
+
+When building for an IDE, the root CMakeLists will group all the files to match their on-disk structure.  This happens automatically.
+
+### Questions
+
+#### Why Add all the directories in each module to the Include Path?
+
+Because we couldn't have duplicate header names anyway.
```
Comment (milesrout) [inline README.md:71]: should that be 'because we _shouldn't_ have duplicate header names anyway'? Because you _can_ have duplicate header names.

**Comment by catageek** [comment]
Is this PR still relevant ?

**Comment by adam4813** [comment]
I don't think it is.

**Comment by DeVaukz** [comment]
Perhaps someday.

**Pull request closed without merging**